src/lib/watt-time/index.ts

```diff
@@ -0,0 +1,134 @@
+import {IImpactModelInterface} from '../interfaces';
+import {KeyValuePair} from '../../types/boavizta';
+import axios from 'axios';
+import * as dayjs from 'dayjs';
+
+export class WattTimeGridEmissions implements IImpactModelInterface {
+  authParams: object | undefined = undefined;
+  token = '';
+  staticParams: object | undefined;
+  name: string | undefined;
+  baseUrl = 'https://api2.watttime.org/v2';
+
+  async authenticate(authParams: object): Promise<void> {
+    this.token = 'token' in authParams ? (authParams['token'] as string) : '';
+    if (this.token.startsWith('ENV_')) {
+      this.token = process.env[this.token.slice(4)] ?? '';
+    }
+    if (this.token === '') {
+      let username =
+        'username' in authParams ? (authParams['username'] as string) : '';
+      let password =
+        'password' in authParams ? (authParams['password'] as string) : '';
+      if (username.startsWith('ENV_')) {
+        username = process.env[username.slice(4)] ?? '';
+      }
+      if (password.startsWith('ENV_')) {
+        password = process.env[password.slice(4)] ?? '';
+      }
+      if (username === '' || password === '') {
+        throw new Error('Missing username or password & token');
+      }
+      const tokenResponse = await axios.get(this.baseUrl + '/login', {
```

Comment:
url can be presented with template literal which is easier to read

`${this.baseUrl}/login`